packages/expo-image-picker/src/ImagePicker.types.ts

```diff
@@ -38,19 +38,26 @@ export type ImageInfo = {
 
 export type ImagePickerResult = { cancelled: true } | ({ cancelled: false } & ImageInfo);
 
+export type ImagePickerMultipleResult = { cancelled: true } | ({ cancelled: false; selected: ImageInfo[] });
+
 export type ImagePickerOptions = {
   allowsEditing?: boolean;
   aspect?: [number, number];
   quality?: number;
-  allowsMultipleSelection?: boolean;
   mediaTypes?: MediaTypeOptions;
   exif?: boolean;
   base64?: boolean;
   videoExportPreset?: VideoExportPreset;
+  allowsMultipleSelection?: boolean;
 };
 
 export type OpenFileBrowserOptions = {
   mediaTypes: MediaTypeOptions;
   capture?: boolean;
   allowsMultipleSelection: boolean;
 };
+
+
+export type ExpandImagePickerResult<T> = T extends ImagePickerOptions & { allowsMultipleSelection: true }
+  ? ImagePickerMultipleResult
+  : ImagePickerResult
```

Comment:
Your code would allow passing into `ExpandImagePickerResult` any js object or primitive.
What I suggest is to narrow down what's accepted in this mapped type to `ImagePickerOptions` (either having `allowsMultipleSelection: true` or not):
```suggestion
export type ExpandImagePickerResult<T extends ImagePickerOptions> = T extends { allowsMultipleSelection: true }
  ? ImagePickerMultipleResult
  : ImagePickerResult
```

Reply:
i got it );